enterprise_access/apps/api/v1/views/subsidy_access_policy.py

```diff
@@ -896,6 +901,103 @@ def can_redeem(self, request, enterprise_customer_uuid):
+            # status.HTTP_404_NOT_FOUND: NotFound,
+        },
+    )
+    @action(
+        detail=False,
+        methods=['get'],
+        url_name='can-request',
+        url_path='enterprise-customer/(?P<enterprise_customer_uuid>[^/.]+)/can-request',
+        pagination_class=None,
+    )
+    def can_request(self, request, enterprise_customer_uuid):
+        """
+        Check if a learner can request access to content. The flow is:
+        1. Find BnR enabled policies first
+        2. Check if content key exists in those policies
+        3. Check for existing pending request by this learner
+        """
+        serializer = serializers.SubsidyAccessPolicyCanRequestRequestSerializer(data=request.query_params)
+        serializer.is_valid(raise_exception=True)
+
+        content_key = serializer.data['content_key']
+        lms_user_id_override = serializer.data.get('lms_user_id') if request.user.is_staff else None
+        lms_user_id = lms_user_id_override or self.lms_user_id or request.user.lms_user_id
+        if not lms_user_id:
+            raise NotFound(detail='Could not determine a value for lms_user_id')
+
+        # Get all active policies for this customer
+        policies_for_customer = self.get_queryset()
+        if not policies_for_customer:
+            raise NotFound(detail='No active policies for this customer')
+
+        # 1. Find policies with BnR enabled
+        bnr_enabled_policies = policies_for_customer.filter(
+            learner_credit_request_config__isnull=False,
+            learner_credit_request_config__active=True
+        )
+        if not bnr_enabled_policies:
+            return Response({
+                'can_request': False,
+                'reason': 'No policies with BnR enabled found'
+            }, status=400)
+
+        # 2. Check if content exists in catalogs for BnR enabled policies
+        # Filter policies to those that contain the content key in their catalog
+        valid_policies = bnr_enabled_policies.filter(
+            pk__in=[
+                policy.pk for policy in bnr_enabled_policies
+                if policy.catalog_contains_content_key(content_key)
+            ]
+        )
+
+        if not valid_policies.exists():
+            return Response({
+                'can_request': False,
+                'reason': REASON_CONTENT_NOT_IN_CATALOG
+            }, status=400)
+
+        # 3. Check for existing pending request by this learner
+        existing_request = LearnerCreditRequest.objects.filter(
+            user__lms_user_id=lms_user_id,
+            enterprise_customer_uuid=enterprise_customer_uuid,
+            course_id=content_key,
+            state__in=[
+                SubsidyRequestStates.REQUESTED,
+                SubsidyRequestStates.APPROVED,
+                SubsidyRequestStates.ACCEPTED,
+                SubsidyRequestStates.ERROR,
+                SubsidyRequestStates.CANCELLED,
+                SubsidyRequestStates.DECLINED,
```

Comment:
A better approach might be to define two separate variables in constants, LC_RE_REQUESTABLE_STATES and LC_NON_RE_REQUESTABLE_STATES, to hold the respective states and then use those in the query.

LC_NON_RE_REQUESTABLE_STATES = [
    SubsidyRequestStates.REQUESTED,
    SubsidyRequestStates.APPROVED,
    SubsidyRequestStates.ACCEPTED,
    SubsidyRequestStates.ERROR,
    SubsidyRequestStates.CANCELLED,
    SubsidyRequestStates.DECLINED,
]

LearnerCreditRequest.objects.filter(
    user__lms_user_id=lms_user_id,
    enterprise_customer_uuid=enterprise_customer_uuid,
    course_id=content_key,
    state__in=LC_NON_RE_REQUESTABLE_STATES
).first()